scripts/tests/test_zkai_stwo_transformer_block_plan.py

```
@@ -0,0 +1,148 @@
+        self.assertEqual(summary["statement_kind"], "transformer-block")
+        self.assertEqual(summary["width"], 4)
+        self.assertEqual(summary["go_criteria_count"], 5)
+        self.assertEqual(summary["no_go_criteria_count"], 4)
+
+    def test_plan_requires_transformer_structure_not_linear_relabel(self) -> None:
+        plan = self._plan()
+        plan["target"]["operations"] = [
+            {"id": "quantized_affine_projection", "description": "linear only"}
+        ]
+
+        with self.assertRaisesRegex(PLAN.PlanValidationError, "rmsnorm_scale_lookup"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)